code/framework/extension/src/main/java/io/cattle/platform/extension/impl/ExtensionManagerImpl.java

```diff
@@ -61,6 +51,21 @@
     }
 
     @Override
+    public Map<String, Object> map(String key) {
+        return getExtensionMapInternal(key);
+    }
+
+    protected ExtensionMap<String, Object> getExtensionMapInternal(String key) {
```

Comment:
Are you sure that this _doesn't_ need to be synchronized? Appears to by modifying state.